Fix hvg subsetting bug

VladimirShitov opened the pull request:
## Changelog

This [line](https://github.com/openpipelines-bio/openpipeline/blob/main/src/integrate/scvi/script.py#L41) in scvi integration component is supposed to subset highly-variable genes:
`adata = adata[:,adata.var['var_input']].copy()`

However, this will not happen for the following reasons:
1. Instead of "var_input" `par["var_input"]` should be used
2. Even with proper key, `adata.var[key]` will return a column from `adata.var` and not the list of selected genes as expected

This PR fixes the issue

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] Conforms to the [Contributor's guide](https://openpipelines.bio/contribute)

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [x] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [ ] CI tests succeed!

DriesSchaumont commented:
@VladimirShitov Good catch! Would you be able to implement tests?

DriesSchaumont reviewed (changes requested):
See comment above

rcannood commented on CHANGELOG.md:
```diff
@@ -32,6 +32,8 @@
 
 * `workflows/full_pipeline`: Fix incorrectly named filtering arguments (#372).
 
+* `integrate/scvi`: Fix subsetting HVG
```
```suggestion
* `integrate/scvi`: Fix bug when subsetting using the `var_input` argument (PR #385).
```

rcannood commented on src/integrate/scvi/script.py:
```diff
@@ -38,7 +38,8 @@ def main():
     if par['var_input']:
         # Subset to HVG
-        adata = adata[:,adata.var['var_input']].copy()
+        selected_genes = adata.var_names[adata.var[par["var_input"]]]
+        adata = adata[:, selected_genes].copy()
```
Would this also work?

```suggestion
        adata = adata[:, adata.var[par["var_input"]]].copy()
```

VladimirShitov replied:
Yes, it should! I over-fixed this

rcannood reviewed (changes requested):
Hi Vladimir,

As mentioned by Dries, would you be able to add a test to the component that checks whether the bug was indeed fixed?

Robrecht

VladimirShitov commented:
Added test and fixed small comments. The function for HVG subsetting is now moved to a separate utils script, as it is used in other components as well. The problem is that the test pipeline fails because there is "No space left on device", so I'm not 100% sure that importing utils works correctly
<img width="1273" alt="image" src="https://github.com/openpipelines-bio/openpipeline/assets/35199218/bf513230-03c5-4bf2-b9bb-6b0b961ab300">

DriesSchaumont commented:
@VladimirShitov The disk issue seemed like a temporary one (?). I am now getting this: 

<img width="1015" alt="Screenshot 2023-06-05 at 09 33 04" src="https://github.com/openpipelines-bio/openpipeline/assets/5946712/e9c27544-fbfa-4617-9234-51b74e0342c4">

VladimirShitov commented:
Yep, this is a problem from my side now 😅 Fixing

VladimirShitov commented:
Fixed it :) @DriesSchaumont , can you check?

DriesSchaumont commented:
@VladimirShitov Could you adress the comment from Robrecht?

rcannood commented on src/utils.py:
```diff
@@ -0,0 +1,16 @@
+def subset_hvg(adata, hvg_col):
+    """Subset highly variable genes from AnnData object
+    
+    Parameters
+    ----------
+    adata : AnnData
+        Annotated data object
+    hvg_col : str
+        Name of the boolean column in `adata.var` that contains the information if genes should be used or not
+
+    Returns
+    -------
+    AnnData
+        Copy of `adata` with subsetted genes
+    """
+    return adata[:, adata.var[hvg_col]].copy()
```
* Rename subset_hvg to subset_vars

Either:
* Move src/utils.py to src/integrate/scvi/helper.py
* Move src/utils.py to src/utils/subset_vars.py

VladimirShitov commented:
Done, please check :)

DriesSchaumont commented:
LGTM! @rcannood If you would like to have a look

DriesSchaumont reviewed (changes requested):
@VladimirShitov Apologies, but some changes were made on `main`. Could you merge again (I will merge this PR as soon as that is done!)
